packages/eslint-plugin/docs/rules/no-floating-promises.mdx

```diff
@@ -115,6 +115,66 @@ await (async function () {
 })();
 ```
 
+### `allowForKnownSafePromises`
+
+This allows you to skip checking of those promises which are marked with some kind of branded type. The branding of a type can be done either by using a type alias or an interface.
```

Comment:
> branded type

[Docs] No, branded types are just one coincidental example of something this option could be used with. A couple more examples:

* `Promise<number> & { hey?: string }`, as in the tests
* `class SafePromise extends Promise`
* `interface SecureThenableLike<T> { then(): T; }`

Reply:
This comment thread isn't resolved 🙂 maybe I wasn't being clear. I'm saying these docs shouldn't say that the `allowForKnownSafePromises` type is for branded types.

Vague phrasing starter:

> This option allows marking specific types as "safe" to be floating.
> For example, you may need to do this in the case of libraries whose APIs return Promises whose rejections are safely handled by the library.

Reply:
ye fixed, forgot to add the docs, in the prev commits they were added as tests, ptal.